Swiften/Elements/MIXParticipant.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <boost/optional.hpp>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/Elements/Payload.h>
+#include <Swiften/Elements/MIXJid.h>
+#include <Swiften/Elements/MIXNick.h>
+
+namespace Swift {
+    class SWIFTEN_API MIXParticipant : public Payload {
+
+        public:
+            typedef std::shared_ptr<MIXParticipant> ref;
+
+        public:
+
+            MIXParticipant() {}
+            
+            const boost::optional<MIXNick>& getNickElement() const {
+                return nick_;
+            }
+
+            void setNickElement(const MIXNick& nick) {
+                nick_ = nick;
+            }
+
+            const boost::optional<MIXJid>& getJIDElement() const {
+                return jid_;
+            }
+
+            void setJIDElement(const MIXJid& jid) {
+                jid_ = jid;
+            }
+
+        private:
+            boost::optional<MIXJid> jid_;
```

Comment:
I wonder if MIXJid and MIXNick elements are needed at all or whether it would make more sense to have an optional JID and optional std::string. The current approach feels a bit overly complex. What do you think @Kev and @intosi ?